packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -1074,6 +1054,42 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
   private isValueEmpty = (value: ShorthandValue | ShorthandCollection) => {
     return _.isArray(value) ? value.length < 1 : !value
   }
+
+  private getHighlightedIndexOnArrowKeyOpen = (
+    changes: StateChangeOptions<ShorthandValue>,
+  ): number => {
+    if (changes.type === Downshift.stateChangeTypes.keyDownArrowUp) {
+      const itemsLength = this.getItemsFilteredBySearchQuery().length
+      return _.isNumber(this.state.highlightedIndex)
+        ? (() => {
+            const newIndex = this.state.highlightedIndex - 1
+            return newIndex < 0 ? itemsLength - 1 : newIndex
+          })()
```

Comment:
What is a actual reason to introduce IIFE there? I also see that it's the same function as on line 1073 🤔

Reply:
will change it, I had it when this was in the onStateChange.